src/lib/mode_handler.js

```diff
@@ -37,15 +37,18 @@ var ModeHandler = function(mode, DrawContext) {
   var delegate = function (eventName, event) {
     var handles = handlers[eventName];
     var iHandle = handles.length;
+    var handlerCalled = false;
     while (iHandle--) {
       var handle = handles[iHandle];
       if (handle.selector(event)) {
         handle.fn.call(ctx, event);
-        DrawContext.store.render();
-        break;
+        handlerCalled = true;
       }
     }
-    DrawContext.ui.updateMapClasses();
+    if (handlerCalled) {
```

Comment:
I'm pretty sure this check is what caused the cursors to go away. Looks like you've fixed that problem by calling render a few times inside of the modes. Either way is good with me as long as the way we go is fastest :).